src/webgpu/api/operation/queue/writeBuffer.spec.ts

```diff
@@ -8,3 +8,89 @@ import { makeTestGroup } from '../../../../common/framework/test_group.js';
 import { GPUTest } from '../../../gpu_test.js';
 
 export const g = makeTestGroup(GPUTest);
+
+g.test('writeBuffer,Ranges')
+  .params([
+    {},
+  ])
+  .fn(async t => {
+    const queue = t.device.defaultQueue;
+
+    function runTest(arrayType : any, testBuffer : boolean) {
+      const elementSize = arrayType.BYTES_PER_ELEMENT;
+      const buffer = t.device.createBuffer({ size: 16 * elementSize, usage: GPUBufferUsage.COPY_DST });
+      let arraySm : any = new arrayType(8);
+      let arrayMd : any = new arrayType(16);
+      let arrayLg : any = new arrayType(32);
+
+      if (testBuffer) {
+        arraySm = arraySm.buffer;
+        arrayMd = arrayMd.buffer;
+        arrayLg = arrayLg.buffer;
+
+        const array15 = new Uint8Array(15).buffer;
+
+        // Writing the full buffer that isn't 4-byte aligned.
+        t.shouldThrow('OperationError', () => queue.writeBuffer(buffer, 0, array15));
+
+        // Writing from an offset that causes source to be 4-byte aligned.
+        t.expectGPUError('validation', () => queue.writeBuffer(buffer, 0, array15, 3), false);
```

Comment:
nit: expectGPUError is not needed for cases where an error is never expected; if there's an error it'll be caught by the "ambient" error checking which occurs around every test. The shouldError param is only needed when it's computed somehow from the params.